src/main/kotlin/nz/co/jedsimson/lgp/examples/kotlin/AntTrail.kt

```diff
@@ -95,24 +95,28 @@ class AntTrailProgram(
     override fun execute() {
         var skipNextInstruction = false
 
-        for (instruction in this.instructions) {
-            if (skipNextInstruction ) {
-                // A branch was not taken so skip the current instruction
-                continue
-            }
-
-            val operation = instruction.operation
+        //run until max moves or ideal fitness
+        while (this.ant.grid.foodRemaining() != 0 && this.ant.state.movesMade < this.ant.maximumMoves) {
```

Comment:
It might be worth adding a kind of sanity-check to this condition in case a degenerate program is created (e.g. one that doesn't actually make any moves). Otherwise this loop may never end.

Reply:
hmm good point. What about checking if movesMade hasn't changed since the last execution? something like `lastMovesMades != movesMade && ...`?

Reply:
Yup, that sounds like a good approach to me.

Reply:
fixed in latest commit